test/Microsoft.AspNetCore.Blazor.E2ETest/Tests/RoutingTest.cs

```diff
@@ -117,6 +117,47 @@ public void CanFollowLinkToOtherPageWithCtrlClick()
             }
         }
 
+        [Fact]
+        public void CanFollowLinkToTargetBlankClick()
+        {
+            try
+            {
+                SetUrlViaPushState("/");
+
+                var app = MountTestComponent<TestRouter>();
+
+                app.FindElement(By.LinkText("Target (_blank)")).Click();
+
+                WaitAssert.Equal(2, () => Browser.WindowHandles.Count);
+            }
+            finally
+            {
+                // Closing newly opened windows if a new one was opened
+                while (Browser.WindowHandles.Count > 1)
+                {
+                    Browser.SwitchTo().Window(Browser.WindowHandles.Last());
+                    Browser.Close();
+                }
+
+                // Needed otherwise Selenium tries to direct subsequent commands
+                // to the tab that has already been closed
+                Browser.SwitchTo().Window(Browser.WindowHandles.First());
+            }
+        }
+
+        [Fact]
+        public void CanFollowLinkToTargetSelfOrParentAndTopClick()
+        {
+            SetUrlViaPushState("/");
+
+            var app = MountTestComponent<TestRouter>();
+            var initialUrl = Browser.Url;
+
+            app.FindElement(By.LinkText("Target (_self)")).Click();
+
+            WaitAssert.Equal(true, () => Browser.WindowHandles.Count == 1 && initialUrl == Browser.Url);
```

Comment:
This doesn't look right. Why are you testing that `initialUrl == Browser.Url`? That's what would already be the case before you clicked the link.

We can't use `WaitAssert` to check for things that are already the case before the action you're testing, because it will evaluate as successful instantly even if your action later proceeds to violate the condition.

Instead we have to use `WaitAssert` to check for something that changes due to the action being tested. For this test, could we have the `target="_self"` link point to some different page, and then have the test check we successfully navigated to it?

Reply:
In this example, target="_ self" performs a page reload.
If there are options how to improve this, suggest, unfortunately there are no options how to test for  target="_ self"

Reply:
As per comment above, I don't think `_self` should trigger a page reload. If there is to be a way of skipping interception for links that open in the same frame, that should be expressed in some other way rather than trying to give a new meaning to `_self`.

Reply:
Forced to agree with you, we can choose the optimal solution for UriHelper.ts and after that based on the new UriHelper.ts I will pass the E2E test for `_self`